src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/RequestForPlacementService.kt

```diff
@@ -0,0 +1,63 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.service
+
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.RequestForPlacement
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.jpa.entity.ApplicationEntity
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.AuthorisableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.transformer.RequestForPlacementTransformer
+import java.util.UUID
+
+@Component
+class RequestForPlacementService(
+  private val applicationService: ApplicationService,
+  private val placementApplicationService: PlacementApplicationService,
+  private val placementRequestService: PlacementRequestService,
+  private val requestForPlacementTransformer: RequestForPlacementTransformer,
+) {
+  fun getRequestsForPlacementByApplication(applicationId: UUID): AuthorisableActionResult<List<RequestForPlacement>> {
+    if (applicationService.getApplication(applicationId) == null) {
+      return AuthorisableActionResult.NotFound("Application", applicationId.toString())
+    }
+
+    val placementApplications = placementApplicationService.getAllPlacementApplicationsForApplicationId(applicationId)
```

Comment:
Assuming the UI isn't filer out any results, I think this is going to return unexpected entities as it will return unsubmitted and reallocated placement apps

We have slightly strange behaviour at the moment where the UI will save unsubmitted placement applications in the backend, but there's no way for it to retrieve them, so they effectively become orphaned if the user navigates away. I don't think we should be returning these applications in this list as i don't think the UI knows what to do with them (although it would be nice to have the ability to managed unsubmitted placement_applications in the future)

Reallocated placement applications shouldn't be returned because they're effectively duplicates of the copy made when reallocation occurs. Ideally we'd handle this with a separate task entity instead of doing this, but that's for the future

I think you should be using the same method as the 'old' endpoint which is now called getAllActivePlacementApplicationsForApplicationId as we know this was returning the correct set of placement applications. Note that this backs onto a repository method that claims to filter out withdrawn placement_applications but that isn't the case as we do see withdrawn request for placements on the current UI. We probably need to tidy up the name on that

Reply:
If you take a look at ApplicaitonTest.GetPlacementApplications there's a bunch of test cases for these scenarios you can probably lift and shift to test the new endpoint